firefox-ios/Client/Frontend/Browser/BrowserViewController/State/BrowserViewControllerState.swift

```diff
@@ -233,6 +236,27 @@ struct BrowserViewControllerState: ScreenState, Equatable {
                 return defaultState(from: state, action: action)
             }
 
+            return BrowserViewControllerState(
+                searchScreenState: state.searchScreenState,
+                windowUUID: state.windowUUID,
+                browserViewType: state.browserViewType,
+                microsurveyState: MicrosurveyPromptState.reducer(state.microsurveyState, action),
+                navigationDestination: NavigationDestination(.homepageZeroSearch)
+            )
+        default:
+            return defaultState(from: state, action: action)
+        }
+    }
+
+    /// Navigate to zero search that shows trending / recent searches state
+    /// after tapping on search button on navigation toolbar
+    static func reduceStateForToolbarAction(
+        action: ToolbarAction,
+        state: BrowserViewControllerState
+    ) -> BrowserViewControllerState {
+        switch action.actionType {
+        case ToolbarActionType.didStartEditingUrl, ToolbarActionType.didDeleteSearchTerm:
```

Comment:
These are the two cases in which we want to trigger showing the zero search state with trending searches. Let me know if there are better actions. `didStartEditingUrl` gives us the general zero search state and the `didDeleteSearchTerm` is when the user has a search term and then deletes it going back to the zero search state.